src/GafferSceneUI/Inspector.cpp

```diff
@@ -91,29 +92,75 @@ Gaffer::Plug *sourceInput( Gaffer::Plug *plug )
 /// return an output plug, such as when it is connected to an anim curve. This
 /// seems to suggest that maybe there are two concepts? Something like a pure
 /// `contextAwareSource()` and a `userEditableSource()`?
-Gaffer::Plug *spreadsheetAwareSource( Gaffer::Plug *plug )
+
+Gaffer::Plug *contextAwareSource( Gaffer::Plug *plug )
 {
 	if( auto sourceValuePlug = plug->source<ValuePlug>() )
 	{
 		if( auto spreadsheet = runTimeCast<Spreadsheet>( sourceValuePlug->node() ) )
 		{
 			if( spreadsheet->outPlug()->isAncestorOf( sourceValuePlug ) )
 			{
-				Gaffer::Plug *inPlug = spreadsheet->activeInPlug( sourceValuePlug );
-				if( inPlug->ancestor<Spreadsheet::RowPlug>() == spreadsheet->rowsPlug()->defaultRow() )
-				{
-					// Don't want to edit the default row, as that could affect
-					// all sorts of other things.
-					return nullptr;
-				}
-				return sourceInput( inPlug );
+				return sourceInput( spreadsheet->activeInPlug( sourceValuePlug ) );
 			}
 		}
 	}
 
 	return sourceInput( plug );
 }
 
+Gaffer::Plug *userEditableSource( Gaffer::Plug *plug )
+{
+	Gaffer::Plug *p = contextAwareSource( plug );
+
+	if( auto spreadsheet = runTimeCast<Spreadsheet>( p->node() ) )
+	{
+		if( p->ancestor<Spreadsheet::RowPlug>() == spreadsheet->rowsPlug()->defaultRow() )
+		{
+			// Don't want to edit the default row, as that could affect
+			// all sorts of other things.
+			return nullptr;
+		}
+	}
+
+	return p;
+}
+
+std::string lockedReason( const Plug *plug, const EditScope *editScope )
+{
+	const GraphComponent *readOnlyReason = MetadataAlgo::readOnlyReason( plug );
+	if( !editScope && readOnlyReason )
+	{
+		return fmt::format(
+			"{} is locked.",
+			readOnlyReason->relativeName( readOnlyReason->ancestor<ScriptNode>() )
+		);
+	}
+	return "";
+}
+
+std::string nonEditableReason( Plug *source, const EditScope *editScope )
+{
+	auto valuePlug = static_cast<const ValuePlug *>( source );
+
+	bool isAnimated = valuePlug ? Animation::isAnimated( valuePlug ) : false;
+	const ValuePlug *sourcePlug = isAnimated ? valuePlug->source<ValuePlug>() : valuePlug;
+
+	std::string result = lockedReason( sourcePlug, editScope );
+	if( result.empty() )
+	{
+		if( !isAnimated && ( sourcePlug && !sourcePlug->settable() ) )
```

Comment:
`ValuePlug::settable()` considers child plugs too, so as written, `nonEditableReason()` only really makes sense for leaf plugs - we don't want to do boolean logic between `isAnimated` (a property of this plug) and `settable()` (potentially a property of a descendant plug). This is why things fell apart when you took out the specific handling for NameValuePlug and TweakPlug.

I suspect things would be a lot easier if `nonEditableReason()` was like `settable()`, and did its own recursion and could reliably be called for compound plugs too.